lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -767,6 +770,23 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif querystr:
+        try:
+            vtype = ipaddr(query, 'type')
+            if vtype == 'address':
+                v = ipaddr(query, 'cidr')
+            elif vtype == 'network':
+                v = ipaddr(query, 'subnet')
+
+            query = netaddr.IPNetwork(v)
+        except:
+            return False
+
+        if network_in_network(query, value):
+            subnetlist = list(query.subnet(value.prefixlen))
+            for i in range(0, len(subnetlist)):
```

Comment:
What you're trying to do is possible to achieve using "sliding window" technique with `enumerate()`, it will be possible to even drop dancing around "index+1" problem:
```python
subnets = list(query.subnet(value.prefixlen))
for i, subnet in enumerate(subnets):
    if subnet == value:
        return str(i+1)
```